packages/components/src/sites-table/use-sites-table-sorting.tsx

```diff
@@ -26,7 +26,7 @@ export function useSitesTableSorting(
 }
 
 function sortSitesByLastPublish( sites: SiteExcerptData[], sortOrder: string ): SiteExcerptData[] {
-	return sites.sort( ( a, b ) => {
+	return [ ...sites ].sort( ( a, b ) => {
```

Comment:
Good catch! Well done keeping the original array unchanged.

Reply:
just for my understanding, @sejas is that the sorting was being applied to the original `allSites` array which intern affects the memoized value?

Reply:
[`[].sort` sorts elements **in place**](http://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)

So a situation could happen like this:

```typescript
// We get site data from somewhere
const sites = [];

// Let's sort the sites (which under the hood mutates the array
const sortedSites = useSitesTableSorting( sites )

// At this point `sites === sortedSites`

// We want to do something whenever the sorted list changes. However this effect is never run! The value we
// pass in the second argument dependency array never changes. As far as React is concerned the sorting
// never changes.
useEffect( () => {
	console.log( 'site list updated' );
	console.log( sites.length );
}, [ sortedSites ] );
```

I don't think this is the issue we had here, but it's why the practice in React isn't to mutate objects (or arrays) in place.

Reply:
@p-jackson Thanks for the explanation!